test/controllers/BankDetailsControllerSpec.scala

```diff
@@ -318,6 +323,67 @@ class BankDetailsControllerSpec extends ItSpec {
       EssttpBackend.ConfirmedDirectDebitDetails.verifyNoneUpdateConfirmDirectDebitDetailsRequest(TdAll.journeyId)
     }
   }
+
+  "GET /terms-and-conditions should" - {
+    "return 200 and the terms and conditions page" in {
+      AuthStub.authorise()
+      EssttpBackend.ConfirmedDirectDebitDetails.findJourneyAfterConfirmUpdateDirectDebitDetails()
+
+      val fakeRequest = FakeRequest().withAuthToken().withSession(SessionKeys.sessionId -> "IamATestSessionId")
+      val result: Future[Result] = controller.termsAndConditions(fakeRequest)
+
+      status(result) shouldBe Status.OK
+      contentType(result) shouldBe Some("text/html")
+      charset(result) shouldBe Some("utf-8")
+
+      val pageContent: String = contentAsString(result)
+      val doc: Document = Jsoup.parse(pageContent)
+
+      doc.title() shouldBe TermsAndConditionsPage.expectedPageTitle
+      doc.select(".govuk-heading-xl").text() shouldBe TermsAndConditionsPage.expectedH1
+      doc.select(".hmrc-header__service-name").text() shouldBe expectedServiceName
+      doc.select(".hmrc-sign-out-nav__link").attr("href") shouldBe "http://localhost:9949/auth-login-stub/session/logout"
+      doc.select("#back").attr("href") shouldBe routes.BankDetailsController.checkBankDetails().url
+
+      val contentItems = List(
+        "We can cancel this agreement if you:",
+        "If we cancel this agreement, you will need to pay the total amount you owe straight away.",
+        "We can use any refunds you might get to pay off your tax charges.",
+        "If your circumstances change and you can pay more or you can pay in full, you need to let us know.",
+        "I agree to the terms and conditions of this payment plan. I confirm that this is the earliest I am able to settle this debt."
+      )
+      doc.select(".govuk-body").asScala.toList
+        .zip(contentItems)
+        .map(things => things._1.text() shouldBe things._2)
+
+      val bulletListItems = List(
+        "pay late or miss a payment",
+        "pay another tax bill late",
+        "do not submit your future tax returns on time"
+      )
+      doc.select(".govuk-list--bullet").select("li").asScala.toList
+        .zip(bulletListItems)
+        .map(things => things._1.text() shouldBe things._2)
```

Comment:
we could probably do a bit better than `things` as a name 😆 Maybe something like this?
```scala
 .map{ case (element, expectedText) => element.text() shouldBe expectedText }
```

Reply:
haha as I was writing it I thought I should do better 😂 , will update